### ISSUE TITLE
Add and register a "flistc:" protocol handler

### ISSUE THREAD

**Issue by alexolog** [fchat-pidgin/fchat-pidgin#145]
Two problems:

1. When I click on a link in a format "`flistc://<something>/<something>`" it is not passed to Pidgin

2. When I try to invoke it manually, using the following command:
`"C:\Program Files\Pidgin\pidgin.exe" --protocolhandler="flistc://<something>/<something>"`
I get the following in the debug log:
```
winpidgin: Got protocol handler request: "flistc://<something>/<something>"
util: Processing message '//<something>/<something>"' for protocol '"flistc' using delimiter '&'.
GLib: g_hash_table_lookup: assertion `hash_table != NULL' failed
GLib: g_hash_table_lookup: assertion `hash_table != NULL' failed
GLib: g_hash_table_lookup: assertion `hash_table != NULL' failed
```

**Comment by vobe** [comment]
A few questions about this request:

* Where exactly do you find flistc:// urls that are not being forwarded to pidgin ? Is it on some web page ? This is intriguing because the first part of those URL is used to locate which account should open the channel in case multiple accounts are logged in
* Assuming flistc:// urls can be found outside pidgin, what should happen when you click on them AND have multiple accounts connected ?

**Comment by alexolog** [comment]
1. I cheated.  Sort of.  I opened the html log in a browser.

2. That is a question I did not think about.  A selection dialog maybe?

**Comment by alexolog** [comment]
From https://wiki.f-list.net/List_of_BBCode_tags : 

> note that you can link to a private channel from a profile on F-List by using the /code command to get your channel's code, then using the [url] tag, above, to create a hyperlink leading to http://www.f-list.net/fchat/?joinChannel=ADH-XXXXXXXXXXXXXXXXXXX -- with the last part, of course, replaced with the ADH code of your channel.

**Comment by vobe** [comment]
I don't think Windows will let us intercept an URL that starts with http:// and forward it to pidgin

**Comment by alexolog** [comment]
Not normally, but there are ways around it:
https://addons.mozilla.org/en-US/firefox/addon/redirector/

**Comment by alexolog** [comment]
Even without http links forwarding, Pudgin registers handlers for aim, msnim, myim, and xmpp protocols, all of them using the --protocolhandler parameter.

**Comment by LurkerHub** [comment]
I encountered the same issue.

There's a link to a chat room on https://www.f-list.net/c/cbe%20slot%20machine
It's https://www.f-list.net/fchat/?joinChannel=ADH-2b5c804cabc154006a57

Clicking on it opens the chat in the browser.

If there was a way to pass the channel to pidgin, I could use an add-on to pass it on.

**Comment by vobe** [comment]
@LurkerHub please try the following build https://github.com/fchat-pidgin/fchat-pidgin/suites/2826019370/artifacts/62958668

In case of multiple accounts, the "first" one (whatever that means) will be joined to the desired channel. This is consistent with what other plugins do.

**Comment by LurkerHub** [comment]
How do I pass the channel to pidgin with that build?

**Comment by vobe** [comment]
> How do I pass the channel to pidgin with that build?

 ```
pidgin.exe --protocolhandler=https://www.f-list.net/fchat/?joinChannel=ADH-2b5c804cabc154006a57
```

**Comment by LurkerHub** [comment]
There's this code in the commit:
```
    // For now, only channel URLs are supported
    // eg https://www.f-list.net/fchat/?joinChannel=ADH-2b5c804cabc154006a57
	if (g_ascii_strcasecmp(proto, "https"))
		return FALSE;
```
Since registering `https` with pidgin is not an option, that prevents opening the link with pidgin simply by clicking it.
If you accept other protocols as well, it will be possible to rewrite the url in the browser to a different scheme.
"flistc" (as mentioned above) seems to be a good option.  Can you accept both?

**Comment by vobe** [comment]
I'm afraid I will need more details than `flistc://<something>/<something>`. Where is the definition of that URL scheme? I could find no examples of such URLs anywhere. Is it something other clients support? If so, an actual example of a `flistc://` URL would be useful

**Comment by LurkerHub** [comment]
I think we are conflating two different issues.

I think that the `flistc:` in the original post referenced an internal identifier.  I can see it inside Pidgin chat windows.  For example when I copy the introduction message to the F-list Sci-Fi channel from the Pidgin window, it contains:
`<b><font size="3"><font color="green">For play and ads that are more magic and fantasy-driven in nature, we suggest the channel <a href="flistc://16a3e7c1/Fantasy">(Channel) Fantasy</a>! The room <a href="flistc://16a3e7c1/World%20of%20Warcraft">(Channel) World of Warcraft</a> is also an option.</font></font></b>`
But the hex identifier changes every time so it is not very useful.

What I asked for is different.

> There's a link to a chat room on https://www.f-list.net/c/cbe%20slot%20machine
> It's https://www.f-list.net/fchat/?joinChannel=ADH-2b5c804cabc154006a57
> 
> Clicking on it opens the chat in the browser.
> If there was a way to pass the channel to pidgin, I could use an add-on to pass it on.

I want to click on a chat link on a web page and have it open in the running instance of Pidgin (or start a new one if not running).
Unfortunately, there isn't a simple way of opening `https` links in external programs.  However, there are browser add-on that allow rewriting urls.  so, for example, it could rewrite `https://www.f-list.net/fchat/?joinChannel=ADH-2b5c804cabc154006a57` to `<something else>://www.f-list.net/fchat/?joinChannel=ADH-2b5c804cabc154006a57` and I could register `<something else>` with `pidgin --protocol-handler` and have it work.

Currently I cannot do it because your code specifically rejects urls that do not start with `https`

**Comment by vobe** [comment]
> But the hex identifier changes every time so it is not very useful.

Indeed, the identifier is computed randomly with a different value every time the plugin is loaded. So these URLs are useless outside of the intended context, which is rendering a clickable link inside Pidgin itself.

> What I asked for is different.

In that case, please open a different issue, and keep in mind that while implementing it there will be no convenient way to select which account opens the channel you clicked on.

**Comment by alexolog** [comment]
`pidgin.exe --protocolhandler=https://www.f-list.net/fchat?joinChannel=ADH-01c9236243213d2af53d` didn't work when Pidgin is already running
